presto-hive/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -82,14 +83,22 @@ public CachingHiveMetastore(@ForCachingHiveMetastore ExtendedHiveMetastore deleg
 
     public CachingHiveMetastore(ExtendedHiveMetastore delegate, ExecutorService executor, Duration cacheTtl, Duration refreshInterval)
     {
-        this.delegate = requireNonNull(delegate, "delegate is null");
+        this(requireNonNull(delegate, "delegate is null"),
+                requireNonNull(executor, "executor is null"),
+                OptionalLong.of(requireNonNull(cacheTtl, "cacheTtl is null").toMillis()),
+                OptionalLong.of(requireNonNull(refreshInterval, "refreshInterval is null").toMillis()));
+    }
+
+    public CachingHiveMetastore(ExtendedHiveMetastore delegate, ExecutorService executor)
```

Comment:
Make this constructor a factory method as well so that it can be apparent to callers that this is a `memoizeMetastore` instead of cache (to emphasize that it doesn't evict based on time/space).